### ISSUE TITLE
[Bug] select2_from_array field with 0 value resets if an request error

### ISSUE THREAD

**Issue by aberaldo** [Laravel-Backpack/CRUD#4099]
# Bug report
This bug is the same as https://github.com/Laravel-Backpack/CRUD/issues/3761 but with select2_from_array.

In a create/update operation if there is a select2_from_array field with value 0 and if there is an error in request form validation, when return the select2_from_array field is empty.
### What I did
In Monsters demo:
1 Go to MonsterCrudController.php 
2 In select2_from_array (no relationship, 1-1 or 1-n) create an option '0' => '0' and save the file.
![image](https://user-images.githubusercontent.com/8792209/122946263-9b851780-d34f-11eb-9e5a-311633e3b868.png)

3 In the panel go to Monsters entity
4 Go to Select tab an set option 0 in select2_from_array (no relationship, 1-1 or 1-n)
5 Click on Save and Back

### What I expected to happen
Show error that The Text field is required and select2_from_array (no relationship, 1-1 or 1-n) with 0 value.
![image](https://user-images.githubusercontent.com/8792209/122946428-ba83a980-d34f-11eb-8b45-807cd3041c3d.png)

### What happened

Show error that The Text field is required and select2_from_array (no relationship, 1-1 or 1-n) with no value.
![image](https://user-images.githubusercontent.com/8792209/122946614-e43cd080-d34f-11eb-82fa-d5e3d9cc05e2.png)

### Is it a bug in the latest version of Backpack?
Yes

**Comment by tabacitu** [comment]
Thank you @aberaldo . Pedro will look at this ASAP - @pxpm please remember to also check 4.2 for this 🙏

**Comment by pxpm** [comment]
Hey @aberaldo !

Thanks for opening the issue. I've sent #4105 to fix it in 4.1. It's fixed in 4.2 without so much if and elses, so clean:

```php
// 4.1

@if((old(square_brackets_to_dots($field['name'])) !== null && (
                        $key == old(square_brackets_to_dots($field['name'])) ||
                        (is_array(old(square_brackets_to_dots($field['name']))) &&
                        in_array($key, old(square_brackets_to_dots($field['name'])))))) ||
                        (null === old(square_brackets_to_dots($field['name'])) &&
                            ((isset($field['value']) && (
                                        $key == $field['value'] || (
                                                is_array($field['value']) &&
                                                in_array($key, $field['value'])
                                                )
                                        )) ||
                                (!isset($field['value']) && isset($field['default']) &&
                                ($key == $field['default'] || (
                                                is_array($field['default']) &&
                                                in_array($key, $field['default'])
                                            )
                                        )
                                ))
                        ))


// 4.2

@if($key == $field['value'] || (is_array($field['value']) && in_array($key, $field['value'])))
```
I am going to close this, I don't think the PR will take much time to merge.

Thanks again,
Pedro

**Comment by tabacitu** [comment]
Merged and tagged! Thank you @pxpm . @aberaldo a `composer update backpack/crud` should do it for you!
